packages/framework/tests/Unit/Model/Payment/IndependentPaymentVisibilityCalculationTest.php

```diff
@@ -0,0 +1,96 @@
+                new DomainConfig(Domain::FIRST_DOMAIN_ID, '', '', 'cs', $defaultTimeZone),
+            );
+
+        $this->customerUserRoleProviderMock = $this->createMock(CustomerUserRoleProvider::class);
+        $this->paymentVisibilityCalculation = new IndependentPaymentVisibilityCalculation(
+            $this->domainMock,
+            $this->customerUserRoleProviderMock,
+        );
+    }
+
+    /**
+     * @param bool $canSeePrices
+     * @param bool $isGatewayPayment
+     * @param bool $isHidden
+     * @param bool $isDeleted
+     * @param bool $isHiddenByGoPay
+     * @param string $name
+     * @param bool $isEnabled
+     * @param bool $expectedResult
+     */
+    #[DataProvider('paymentVisibilityProvider')]
+    public function testIsIndependentlyVisible(
+        bool $canSeePrices,
+        bool $isGatewayPayment,
+        bool $isHidden,
+        bool $isDeleted,
+        bool $isHiddenByGoPay,
+        string $name,
+        bool $isEnabled,
+        bool $expectedResult,
+    ) {
+        $paymentMock = $this->createMock(Payment::class);
+        $paymentMock->method('isGatewayPayment')->willReturn($isGatewayPayment);
+        $paymentMock->method('isHidden')->willReturn($isHidden);
+        $paymentMock->method('isDeleted')->willReturn($isDeleted);
+        $paymentMock->method('isHiddenByGoPay')->willReturn($isHiddenByGoPay);
+        $paymentMock->method('getName')->willReturn($name);
+        $paymentMock->method('isEnabled')->willReturn($isEnabled);
+
+        $this->customerUserRoleProviderMock->method('canCurrentCustomerUserSeePrices')->willReturn($canSeePrices);
+        $this->domainMock->method('getDomainConfigById')->willReturn((object)['locale' => 'en']);
+
+        $this->assertEquals($expectedResult, $this->paymentVisibilityCalculation->isIndependentlyVisible($paymentMock, 1));
+    }
+
+    /**
+     * @return array
+     */
+    public static function paymentVisibilityProvider(): array
+    {
+        return [
+            'Customer can see prices' => [true, false, false, false, false, 'Payment Name', true, true],
+            'Payment name is empty' => [true, false, false, false, false, '', true, false],
+            'Payment is hidden' => [true, false, true, false, false, 'Payment Name', true, false],
+            'Payment is deleted' => [true, false, false, true, false, 'Payment Name', true, false],
+            'Payment is hidden by GoPay' => [true, false, false, false, true, 'Payment Name', true, false],
+            'Payment is enabled' => [true, false, false, false, false, 'Payment Name', true, true],
+            'Payment is not enabled' => [true, false, false, false, false, 'Payment Name', false, false],
+            'Customer cannot see prices and payment is a gateway payment' => [false, true, false, false, false, 'Payment Name', true, false],
+            'Customer cannot see prices and payment is not a gateway payment' => [false, false, false, false, false, 'Payment Name', true, true],
```

Comment:
could you please use associative arrays in the provider? It helps the readability a lot :pray: 
```php
        return [
            'Customer can see prices' => [
                'canSeePrices' => true,
                'isGatewayPayment' => false,
                'isHidden' => false,
                'isDeleted' => false,
                'isHiddenByGoPay' => false,
                'name' => 'Payment Name',
                'isEnabled' => true,
                'expectedResult' => true
            ],
// ...
```

Reply:
fixed

Reply:
https://github.com/shopsys/shopsys/pull/3355/commits/9670bc675188f75080148f384f0d2698c1cdaf54